src/addr.rs

```diff
@@ -258,6 +259,17 @@ impl AddAssign<u64> for VirtAddr {
     }
 }
 
+impl CheckedAdd<u64> for VirtAddr {
+    type Output = Self;
+
+    fn checked_add(self, rhs: u64) -> Option<Self::Output> {
+        self.0
+            .checked_add(rhs)
+            .and_then(is_canonical)
+            .map(Self::new)
```

Comment:
This can just be:
```rust
from_canonical(self.0.checked_add(rhs)?)
```
Ideally it would be:
```rust
VirtAddr::try_new(self.0.checked_add(rhs)?).ok()
```
but that doesn't work see #299